piccolo/query/mixins.py

```diff
@@ -237,6 +238,49 @@ def copy(self) -> OutputDelegate:
         return self.__class__(_output=_output)
 
 
+class SyncCallback(t.Protocol):
+    def __call__(self, success: bool, results: t.Any):
+        ...
+
+
+class AsyncCallback(t.Protocol):
+    async def __call__(self, success: bool, results: t.Any):
+        ...
+
+
+Callback = t.Union[SyncCallback, AsyncCallback]
+
+
+@dataclass
+class CallbackDelegate:
+    """
+    Example usage:
+
+    .callback(my_handler_function)
+    .callback(my_handler_coroutine)
+    .callback(lambda success, row: print(row))
+    """
+
+    _callback: t.Optional[Callback] = None
+
+    def callback(self, callback: Callback):
+        self._callback = callback
```

Comment:
I think it would be cool if we could add multiple callbacks, So maybe it's `add_callback`, and `_callback` is a list.